edx/edx-enterprise
# ENT-850 Indicate when a course is no longer open for enrollment

@asadiqbal08 @saleem-latif 
**Description:** Indicate when a course is no longer open for enrollment by adding the message `Enrollment Closed` in the title along with the start date with the format `edX Demonstration Course (February 2013 - Enrollment Closed)`

**JIRA:** [ENT-850](https://openedx.atlassian.net/browse/ENT-850)

**Dependencies:** N/A

**Merge deadline:** 9 Feb, 2018

**Installation instructions:**  Install `edx-enterprise` from this branch `zub/ENT-850-enrollment-closed-message-in-course-title` in `edx-platform`.

**Testing instructions:**

1. Add an enterprise with catalogs
2. Add configurations for integrated channels for `Degreed` and `SAP Success Factors`
3. Add course run A in above catalog with [availability `Current`](https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L724-L736) 
3. Add course run B in above catalog with [availability `Starting Soon`](https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L724-L736) 
3. Add course run C in above catalog with [availability `Upcoming`](https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L724-L736)
4. Add course run D in above catalog with [availability `Archived`](https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L724-L736)
4. Run django management command `transmit_course_metadata` and verify that course runs A, B and C have start date in their title with the format `edX Demonstration Course (Starts: February 2013)` and course D with availability `Archived` have start date in its title with the format `edX Demonstration Course (February 2013 - Enrollment Closed)`

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
